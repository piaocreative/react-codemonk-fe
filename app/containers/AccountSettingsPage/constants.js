export const NOTIFICATION_LIST = [
  'New jobs alert',
  'Interview requests',
  'Timesheet approvals',
  'Payment notifications',
  'New product features',
  'Marketing newsletters',
];

export const PREFERENCES = ['Email', 'SMS', 'Calls'];
