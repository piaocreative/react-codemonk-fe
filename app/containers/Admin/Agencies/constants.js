/*
 * AdminAgencies Constants
 */
import React from 'react';
import { TableSkeletonCol6 } from 'components/SkeletonLoader';

export const key = 'AdminAgencies';
export const CHANGE_AGENCY_STATUS = 'CodeMonk/AdminAgencies/CHANGE_AGENCY_STATUS';
export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Name',
    selector: 'name',
  },
  {
    name: 'Email',
    selector: 'email',
  },
  {
    name: 'Status',
    selector: 'status',
  },
  {
    name: 'Action',
    selector: 'action',
    allowOverflow: true,
    style: {
      paddingRight: '40px',
    },
  },
];

export const agencyActionStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Active',
    value: 1,
  },
  {
    label: 'Suspend',
    value: 2,
  },
  {
    label: 'Unregistered',
    value: 0,
  },
];

export const LinearIndeterminate = () => (
  <div className="w-100 flex-column d-flex">
    <TableSkeletonCol6 cardCount={5} />
  </div>
);
