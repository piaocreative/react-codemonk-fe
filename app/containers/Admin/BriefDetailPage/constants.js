export const adminBriefTalentListingColumns = [
  {
    name: 'Name',
    selector: 'name',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Role',
    selector: 'role',
  },
  {
    name: 'Rate',
    selector: 'rate',
    style: {
      paddingRight: '40px',
    },
  },
];
