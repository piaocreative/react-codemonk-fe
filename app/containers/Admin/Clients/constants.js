/*
 * AdminClients Constants
 */

export const key = 'AdminClients';
export const CHANGE_CLIENT_STATUS = 'CodeMonk/AdminClients/CHANGE_STATUS';
export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Name',
    selector: 'name',
  },
  {
    name: 'Company Name',
    selector: 'compnayName',
  },
  {
    name: 'Email',
    selector: 'email',
  },
  {
    name: 'Status',
    selector: 'status',
  },
  {
    name: 'Action',
    selector: 'action',
    allowOverflow: true,
    style: {
      paddingRight: '40px',
    },
  },
];

export const clientActionStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Active',
    value: 1,
  },
  {
    label: 'Suspend',
    value: 2,
  },
  {
    label: 'Unregistered',
    value: 0,
  },
];
