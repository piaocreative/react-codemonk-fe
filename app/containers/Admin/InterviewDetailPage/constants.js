export const key = 'AdminInterviewDetail';

export const CHANGE_TALENT_STATUS = 'CodeMonk/AdminInterviewDetail/CHANGE_TALENT_STATUS';

export const talentStatusArray = [
  {
    label: 'Hired',
    value: 1,
  },
  {
    label: 'Rejected',
    value: 0,
  },
];
