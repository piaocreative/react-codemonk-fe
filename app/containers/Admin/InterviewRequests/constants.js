/*
 * AdminInterviews Constants
 */

import React from 'react';
import { TableSkeletonCol6 } from 'components/SkeletonLoader';

export const key = 'AdminInterviews';

export const CHANGE_PROJECTS_ARRAY = 'CodeMonk/AdminInterviews/CHANGE_PROJECTS_ARRAY';
export const CHANGE_INTERVIEW_STATUS = 'CodeMonk/AdminInterviews/CHANGE_INTERVIEW_STATUS';

export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const projectStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Requested',
    value: 0,
  },
  {
    label: 'In Progress',
    value: 1,
  },
  {
    label: 'Done',
    value: 2,
  },
];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Date Requested',
    selector: 'dateRequested',
  },
  {
    name: 'Client Name',
    selector: 'clientName',
  },
  {
    name: 'Company Name',
    selector: 'companyName',
  },
  {
    name: 'Project Name',
    selector: 'name',
  },
  {
    name: 'Talent Name',
    selector: 'talentName',
  },
  {
    name: 'Status',
    selector: 'status',
    minWidth: '150px',
    allowOverflow: true,
  },
  {
    name: '',
    selector: 'action',
    maxWidth: '100px',
    style: {
      paddingRight: '40px',
    },
  },
];

export const LinearIndeterminate = () => (
  <div className="w-100 flex-column d-flex">
    <TableSkeletonCol6 cardCount={5} />
  </div>
);
