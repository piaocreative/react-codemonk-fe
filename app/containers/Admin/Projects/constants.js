/*
 * AdminProjects Constants
 */

export const key = 'AdminProjects';
export const FILTER_KEYS = ['status'];

export const DEFAULT_PAGE_NO = 1;
export const CHANGE_PROJECTS_ARRAY = 'CodeMonk/AdminProjects/CHANGE_PROJECTS_ARRAY';
export const CHANGE_PROJECT_STATUS = 'CodeMonk/AdminProjects/CHANGE_STATUS';
export const ADD_PROJECT = 'CodeMonk/AdminProjects/ADD_PROJECT';

export const projectStatusPopup = [
  {
    label: 'Proposed',
    value: '1',
  },
  {
    label: 'Discovery',
    value: '2',
  },
  {
    label: 'Kick-off',
    value: '3',
  },
  {
    label: 'In Progress',
    value: '4',
  },
  {
    label: 'On Hold',
    value: '5',
  },
  {
    label: 'Suspended',
    value: '6',
  },
  {
    label: 'Closed',
    value: '7',
  },
];

const adminProjectsColumns1 = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Project Name',
    selector: 'name',
  },
  {
    name: 'Client Name',
    selector: 'clientName',
  },
  {
    name: 'Company Name',
    selector: 'companyName',
  },
];
const adminProjectsColumns2 = [
  {
    name: 'Start Date',
    selector: 'startDate',
  },
  {
    name: 'End Date',
    selector: 'endDate',
  },
  {
    name: 'Status',
    selector: 'status',
    maxWidth: '165px',
    allowOverflow: true,
  },
  {
    name: '',
    selector: 'action',
    style: {
      paddingRight: '40px',
      justifyContent: 'center',
    },
  },
];
export const adminProjectsColumns = [...adminProjectsColumns1, ...adminProjectsColumns2];
