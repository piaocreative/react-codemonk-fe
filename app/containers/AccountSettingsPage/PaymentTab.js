/** PersonalId Page
 * This is the PersonalId page for the App, at the '/my-profile' route
 */
import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { FormGroup, Row, Col } from 'reactstrap';
import get from 'lodash/get';
import { H4, LinkButtonMod, Button, FormWrapper, FormLabel, P } from 'components';
import { createStructuredSelector } from 'reselect';
import { Field, reduxForm, change } from 'redux-form/immutable';
import * as formValidations from 'utils/formValidations';
import * as normalize from 'utils/normalize';
import { FormattedMessage } from 'react-intl';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import { loadRepos } from 'containers/App/actions';
import { RenderRadio, renderFieldPassword } from 'utils/Fields';
import { makeSelectLoading, makeSelectSuccess, makeSelectError } from 'containers/App/selectors';
import * as actions from 'containers/Auth/PaymentAndBilling/actions';
import * as selectors from 'containers/Auth/PaymentAndBilling/selectors';
import saga from 'containers/Auth/PaymentAndBilling/saga';
import reducer from 'containers/Auth/PaymentAndBilling/reducer';
import { key } from 'containers/Auth/PaymentAndBilling/constants';
import { propTypes } from 'containers/proptypes';
import containerMessage from 'containers/messages';
import authMessages from 'containers/Auth/PaymentAndBilling/messages';
import { PaymentPageWrapper } from 'containers/Auth/PaymentAndBilling/payment-styles';
import { Payment } from 'containers/Auth/PaymentAndBilling/utils';

export class PaymentTab extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      passwordType: 'password',
    };
  }

  componentDidMount() {
    this.initilizeData();
  }

  initilizeData = () => {
    const { dispatch, data } = this.props;
    const payType = get(data, 'pay.type', '');

    const bankName = get(data, 'pay.bankDetails.name', '');
    const bankAccountNumber = get(data, 'pay.bankDetails.accNumber', '');
    const bankCode = get(data, 'pay.bankDetails.bankCode', '');
    const bankPayDetails = {
      bankAccountNumber,
      bankCode,
      bankName,
    };
    const payPalEmail = get(data, 'pay.payPalEmail', '');

    dispatch(change(key, 'payType', payType));
    dispatch(change(key, 'bankName', bankName));
    dispatch(change(key, 'bankAccountNumber', bankAccountNumber));
    dispatch(change(key, 'bankCode', bankCode));
    dispatch(change(key, 'password', ''));
    dispatch(change(key, 'payPalEmail', payPalEmail));

    dispatch(actions.changePayType(payType));
    dispatch(actions.changeBankPayDetails(bankPayDetails));
    dispatch(actions.changePayPalDetails(payPalEmail));
  };

  handleChangePayType = e => {
    const { dispatch, onChangePayType } = this.props;
    const { value } = e.target;
    dispatch(change(key, 'payType', value));
    onChangePayType(value);
  };

  handleChangeBankPayDetails = e => {
    const { dispatch, onChangeBankPayDetails, bankPayDetails } = this.props;
    const { value, name } = e.target;
    dispatch(change(key, name, value));
    const newbankPayDetails = bankPayDetails;
    newbankPayDetails[name] = value;
    onChangeBankPayDetails(newbankPayDetails);
  };

  handleChangePayPalDetails = e => {
    const { value } = e.target;
    const { dispatch, onChangePayPalDetails } = this.props;
    dispatch(change(key, 'payPalEmail', value));
    onChangePayPalDetails(value);
  };

  handleSubmitButton = invalid => {
    let output = true;
    if (!invalid) {
      output = false;
    }
    return output;
  };

  onClickIcon = type => {
    if (type === 'passwordType') {
      this.setState(({ passwordType }) => ({
        passwordType: passwordType === 'text' ? 'password' : 'text',
      }));
    }
  };

  handlePaymentSubmit = e => {
    const { onSubmitPayment } = this.props;
    onSubmitPayment(e, this.paymentSubmitSuccess);
  };

  paymentSubmitSuccess = () => {
    const { loadData } = this.props;
    loadData();
  };

  render() {
    const { handleSubmit, loading, invalid, payType, bankPayDetails, payPalEmail, onChangePassword } = this.props;
    const { passwordType } = this.state;

    return (
      <PaymentPageWrapper>
        <FormWrapper>
          <form onSubmit={handleSubmit}>
            <H4 className="newH4 mt-4" opacityVal="0.5">
              <FormattedMessage {...authMessages.labelPayment} />
            </H4>

            <FormGroup>
              <Field
                name="payType"
                component={RenderRadio}
                data={Payment(payType, bankPayDetails, this.handleChangeBankPayDetails, payPalEmail, this.handleChangePayPalDetails)}
                groupName="payment"
                onChangeRadio={e => this.handleChangePayType(e)}
                selectedRadio={payType}
                validate={[formValidations.required]}
                editFlag={false}
              />
            </FormGroup>
            <Row>
              <Col md={6} sm={12}>
                <H4 className="newH4 mb-3  mt-5" opacityVal="0.5">
                  <FormattedMessage {...containerMessage.confirmChanges} />
                </H4>
                <P className="p16 mb-3" opacityVal="0.5">
                  <FormattedMessage {...containerMessage.passwordDescription} />
                </P>
                <FormGroup>
                  <FormLabel>
                    <FormattedMessage {...containerMessage.tabPassword} />
                  </FormLabel>
                  <Field
                    name="password"
                    type={passwordType}
                    component={renderFieldPassword}
                    onChange={onChangePassword}
                    placeholder={containerMessage.placeholderPasswordAddress.defaultMessage}
                    normalize={normalize.trimSpace}
                    validate={[formValidations.required]}
                    onClick={() => {
                      this.onClickIcon('passwordType');
                    }}
                  />
                </FormGroup>
              </Col>
            </Row>
            <div className="d-flex align-items-center flex-wrap justify-content-between justify-content-md-start mt-5">
              <Button
                className={`${loading ? 'loading' : ''} btn-primary me-0 mt-0 mr-3`}
                disabled={this.handleSubmitButton(invalid)}
                onClick={handleSubmit(e => this.handlePaymentSubmit(e))}
              >
                <FormattedMessage {...containerMessage.btnUpdate} />
              </Button>
              <LinkButtonMod color="link" className="me-3 ms-md-3" onClick={this.initilizeData}>
                <FormattedMessage {...containerMessage.btnCancel} />
              </LinkButtonMod>
            </div>
          </form>
        </FormWrapper>
      </PaymentPageWrapper>
    );
  }
}

PaymentTab.propTypes = propTypes;

const mapStateToProps = createStructuredSelector({
  payType: selectors.makeSelectPayType(),
  bankPayDetails: selectors.makeSelectBankPayDetails(),
  payPalEmail: selectors.makeSelectPayPalEmail(),
  loading: makeSelectLoading(),
  responseSuccess: makeSelectSuccess(),
  responseError: makeSelectError(),
  password: selectors.makeSelectPassword(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onChangePayType: data => dispatch(actions.changePayType(data)),
    onChangeBankPayDetails: data => dispatch(actions.changeBankPayDetails(data)),
    onChangePayPalDetails: data => dispatch(actions.changePayPalDetails(data)),
    onChangePassword: event => dispatch(actions.passwordAction(event.target.value)),
    onSubmitPayment: (evt, onSuccess) => {
      if (evt !== undefined && evt.preventDefault) {
        evt.preventDefault();
      }
      dispatch(loadRepos());
      dispatch(actions.editPayment(onSuccess));
    },
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);
const withReducer = injectReducer({ key, reducer });
const withSaga = injectSaga({ key, saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
  reduxForm({
    form: key,
    touchOnChange: true,
  }),
)(PaymentTab);
