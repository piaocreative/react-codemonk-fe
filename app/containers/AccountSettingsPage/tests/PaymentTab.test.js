import React from 'react';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import request from 'utils/request';
import { LOAD_REPOS } from 'containers/App/constants';
import initialState from 'containers/Auth/PaymentAndBilling/reducer';
import { CHANGE_PAY_TYPE, CHANGE_BANK_PAY_DETAILS, CHANGE_PAYPAL_DETAILS, EDIT_PAYMENT } from 'containers/Auth/PaymentAndBilling/constants';
import { PaymentTab as MainForm, mapDispatchToProps } from '../PaymentTab';
jest.mock('utils/request');
let store;
const mockStore = configureStore();
const props = {
  payType: '',
  bankPayDetails: {},
  payPalEmail: '',

  dispatch: jest.fn(),
  handleSubmit: jest.fn(),

  invalid: false,
  loading: false,
  responseSuccess: false,
  responseError: false,
  history: { push: jest.fn(), replace: jest.fn() },
  location: {},

  onChangePayType: jest.fn(),
  onChangeBankPayDetails: jest.fn(),
  onChangePayPalDetails: jest.fn(),
  onSubmitPayment: jest.fn(),
};
const getWrapper = () => shallow(<MainForm store={store} {...props} />);
const getInstance = () => getWrapper().instance();
const intializeSetup = () => {
  store = mockStore(initialState);
  request.mockImplementation(() => Promise.resolve({ status: 1 }));
};

describe('<MainForm />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<MainForm store={store} {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});

describe('test functions', () => {
  intializeSetup();
  getWrapper();

  test('Testing if handleChangePayType', () => {
    const handleChangePayType = jest.spyOn(getInstance(), 'handleChangePayType');
    handleChangePayType({ target: { name: 'payType', value: 'paypal' } });
    expect(handleChangePayType).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleChangeBankPayDetails', () => {
    const handleChangeBankPayDetails = jest.spyOn(getInstance(), 'handleChangeBankPayDetails');
    handleChangeBankPayDetails({ target: { name: 'bankName', value: 'ASBCD Bank' } });
    expect(handleChangeBankPayDetails).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleChangePayPalDetails', () => {
    const handleChangePayPalDetails = jest.spyOn(getInstance(), 'handleChangePayPalDetails');
    handleChangePayPalDetails({ target: { name: 'paypalEmail', value: 'user@paypal.com' } });
    expect(handleChangePayPalDetails).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleSubmitButton ', () => {
    const handleSubmitButton = jest.spyOn(getInstance(), 'handleSubmitButton');
    handleSubmitButton(false, true);
    expect(handleSubmitButton).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleSubmitButton with else', () => {
    const handleSubmitButton = jest.spyOn(getInstance(), 'handleSubmitButton');
    handleSubmitButton(true, true);
    expect(handleSubmitButton).toHaveBeenCalledTimes(1);
  });
});

describe('mapDispatchToProps Testing', () => {
  intializeSetup();
  getWrapper();
  test('Testing if  CHANGE_PAY_TYPE are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onChangePayType('freelancer');
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: 'freelancer',
      type: CHANGE_PAY_TYPE,
    });
  });

  test('Testing if  CHANGE_BANK_PAY_DETAILS are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onChangeBankPayDetails({});
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: {},
      type: CHANGE_BANK_PAY_DETAILS,
    });
  });

  test('Testing if  CHANGE_PAYPAL_DETAILS are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onChangePayPalDetails({});
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: {},
      type: CHANGE_PAYPAL_DETAILS,
    });
  });

  const onSuccess = jest.fn();
  test('Testing if the  LOAD_REPOS events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProps(dispatch).onSubmitPayment(event, onSuccess);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: LOAD_REPOS,
    });
  });

  test('Testing if the  EDIT_BILLING events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProps(dispatch).onSubmitPayment(event, onSuccess);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: EDIT_PAYMENT,
      onSuccess,
    });
  });
});
