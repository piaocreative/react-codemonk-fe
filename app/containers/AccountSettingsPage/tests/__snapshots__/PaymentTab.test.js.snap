// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> should render and match the snapshot 1`] = `
<ForwardRef(payment-styles__PaymentPageWrapper)>
  <ForwardRef(FormWrapper)>
    <form
      onSubmit={
        [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
            Array [
              [Function],
            ],
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <ForwardRef(H4)
        className="newH4 mt-4"
        opacityVal="0.5"
      >
        <FormattedMessage
          defaultMessage="Payout via"
          id="CodeMonk.components.PaymentAndBilling.labelPayment"
          values={Object {}}
        />
      </ForwardRef(H4)>
      <FormGroup
        tag="div"
      >
        <ForwardRef
          component={[Function]}
          data={
            Array [
              Object {
                "Component": <BankAccountDetails
                  bankPayDetails={Object {}}
                  editFlag={true}
                  handleChangeBankPayDetails={[Function]}
                />,
                "className": "mb-3",
                "condition": false,
                "label": "Bank Account",
                "value": "bank",
              },
              Object {
                "Component": <PayPalDetails
                  editFlag={true}
                  handleChangePayPalDetails={[Function]}
                  payPalEmail=""
                />,
                "className": "mb-3",
                "condition": false,
                "label": "Paypal",
                "value": "paypal",
              },
            ]
          }
          editFlag={false}
          groupName="payment"
          name="payType"
          onChangeRadio={[Function]}
          selectedRadio=""
          validate={
            Array [
              [Function],
            ]
          }
        />
      </FormGroup>
      <Row
        tag="div"
      >
        <Col
          md={6}
          sm={12}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <ForwardRef(H4)
            className="newH4 mb-3  mt-5"
            opacityVal="0.5"
          >
            <FormattedMessage
              defaultMessage="Confirm changes"
              id="CodeMonk.containers.containerMessage.confirmChanges"
              values={Object {}}
            />
          </ForwardRef(H4)>
          <ForwardRef(P)
            className="p16 mb-3"
            opacityVal="0.5"
          >
            <FormattedMessage
              defaultMessage="Please enter your password to confirm these changes"
              id="CodeMonk.containers.containerMessage.passwordDescription"
              values={Object {}}
            />
          </ForwardRef(P)>
          <FormGroup
            tag="div"
          >
            <ForwardRef(Label__FormLabel)>
              <FormattedMessage
                defaultMessage="Password"
                id="CodeMonk.containers.containerMessage.tabPassword"
                values={Object {}}
              />
            </ForwardRef(Label__FormLabel)>
            <ForwardRef
              component={[Function]}
              name="password"
              normalize={[Function]}
              onClick={[Function]}
              placeholder="Enter your password"
              type="password"
              validate={
                Array [
                  [Function],
                ]
              }
            />
          </FormGroup>
        </Col>
      </Row>
      <div
        className="d-flex align-items-center flex-wrap justify-content-between justify-content-md-start mt-5"
      >
        <Button
          className=" btn-primary me-0 mt-0 mr-3"
          disabled={false}
        >
          <FormattedMessage
            defaultMessage="Update"
            id="CodeMonk.containers.containerMessage.btnUpdate"
            values={Object {}}
          />
        </Button>
        <ForwardRef(LinkButtonMod__ButtonMod)
          className="me-3 ms-md-3"
          color="link"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Cancel"
            id="CodeMonk.containers.containerMessage.btnCancel"
            values={Object {}}
          />
        </ForwardRef(LinkButtonMod__ButtonMod)>
      </div>
    </form>
  </ForwardRef(FormWrapper)>
</ForwardRef(payment-styles__PaymentPageWrapper)>
`;
